### PR TITLE
Update dependencies to PHP 7.1 and PHPUnit 7

### PR DESCRIPTION
**Pull request by robbieaverill**
To fix the master (SS5) builds.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,8 +30,9 @@ before_script:
 
 # Install composer dependencies
   - composer validate
-  - composer require --no-update silverstripe/recipe-cms:1.0.x-dev
+  - composer require --no-update silverstripe/recipe-cms:2.x-dev
   - composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
+  - composer require --prefer-source silverstripe/framework 5.x-dev
```
Comment (tractorcow) [inline .travis.yml]: Shouldn't this be before the `composer install`, and also have --no-update?

Comment (robbieaverill) [reply .travis.yml]: I don't know - can you determine in whether to install from source or dist before you run a "install from dist" command? I expected that this flag would be negated by the use of `--no-update`

Comment (tractorcow) [reply .travis.yml]: Sorry, turns out it didn't work. `Nothing to install or update`

**Comment by tractorcow** [comment]
@robbieaverill you'll just need to --prefer-source instead of dist.

**Comment by robbieaverill** [comment]
You mean for the whole “composer install” command?

**Comment by robbieaverill** [comment]
Updated

**Pull request merged**